### ISSUE TITLE
Tests are failing with "type 'JsObjectImpl' is not a subtype of type 'List<String>' of 'function result'." since 1.13.x

### ISSUE THREAD

**Issue by MikeMitterer** [dart-lang/sdk#24806]
I'm running my tests with `pub run test -p content-shell test/unit/test.dart``

With 1.12.1 all my tests are running fine but with 1.13.0-dev.7.x I get 

```bash
00:03 +2 -5: RetValue mit Futures > Secret > mit falschen Secret
  type 'JsObjectImpl' is not a subtype of type 'List<String>' of 'function result'.
  dart:indexed_db                                              Database.objectStoreNames
  package:lawndart/src/indexeddb_store.dart 51:13              IndexedDbStore._open.<async>
  ===== asynchronous gap ===========================
  dart:async                                                   _Completer.completeError
  package:lawndart/src/indexeddb_store.dart 67:3               IndexedDbStore._open.<async>
  ===== asynchronous gap ===========================
  dart:async                                                   _asyncThenWrapperHelper
  package:lawndart/src/indexeddb_store.dart                    IndexedDbStore._open
  package:lawndart/lawndart.dart 82:17                         Store.open.<async>
  ===== asynchronous gap ===========================
  dart:async                                                   Future.Future.microtask
  package:lawndart/lawndart.dart                               Store.open
  package:webapp_base_ui/src/persistance/ApiKeyDAO.dart 79:48  ApiKeyDAO._db._db_async_body>
  ===== asynchronous gap ===========================
  dart:async                                                   Future.Future.microtask
  package:webapp_base_ui/src/persistance/ApiKeyDAO.dart        ApiKeyDAO._db
  package:webapp_base_ui/src/persistance/ApiKeyDAO.dart 71:32  ApiKeyDAO.removeCredentials.<async>
  ===== asynchronous gap ===========================
  dart:async                                                   Future.Future.microtask
  package:webapp_base_ui/src/persistance/ApiKeyDAO.dart        ApiKeyDAO.removeCredentials
  persistance/apikey_test.dart 29:21                           testApiKey.<fn>.<fn>

  type '_TypeError' is not a subtype of type 'String' of 'errormessage'.
  persistance/apikey_test.dart 143:83                          testApiKey.<fn>.<fn>.<fn>.<fn>
  dart:async                                                   _Completer.completeError
  package:webapp_base_ui/src/persistance/ApiKeyDAO.dart 119:5  ApiKeyDAO._setKey.<async>
  dart:async                                                   _Completer.completeError
  package:webapp_base_ui/src/persistance/ApiKeyDAO.dart 79:88  ApiKeyDAO._db._db_async_body>
  dart:async                                                   _Completer.completeError
  package:lawndart/lawndart.dart 88:3                          Store.open.<async>
  dart:async                                                   _Completer.completeError
  package:lawndart/src/indexeddb_store.dart 67:3               IndexedDbStore._open.<async>

00:03 +22 -5: Some tests failed.
mikemitterer@Mikes-Mac-Pro-2 WebApp.Base.UI $
mikemitterer@Mikes-Mac-Pro-2 WebApp.Base.UI $ brew switch dart 1.12
mikemitterer@Mikes-Mac-Pro-2 WebApp.Base.UI $ dart --version
Dart VM version: 1.12.1 (Tue Sep  8 04:10:39 2015) on "macos_x64"
mikemitterer@Mikes-Mac-Pro-2 WebApp.Base.UI $ pub run test -p content-shell test/unit/test.dart
Wrong script snapshot version, expected '6d2168eae45307fb28901b56b5c19b62' found '0c79e71b1b4f251f483c790bbacdd406'
00:03 +27: All tests passed!
```

**Comment by zoechi** [comment]
What Dart, content-shell version are you using. Does it work with Dartium, Chrome, Firefox?

**Comment by MikeMitterer** [comment]
It works in Dartium and if compiled to JS it works with all major browsers. That's not the problem - "only"??? the tests with content-shell are failing.

Content-Shell is 1.13.0-dev.7.6 but was the same with 1.13.0-dev.7.4

**Comment by kevmoo** [comment]
Can we get looked at ASAP for 1.13?

CC @sigmundch @jacob314 @dgrove

**Comment by alan-knight** [comment]
Do you have a repro? When it fails, is it always on objectStoreNames, or are there other failures?

This basically looks like it's failing to correctly wrap the JS Object that comes back in the corresponding Dart object.

**Comment by alan-knight** [comment]
Yes, we don't wrap JsArray, since it's usable as a Dart List, but we're doing the test based on the Dart Type. This returns a DomStringList, which maps to a Dart List<String>, but isn't a JsArray. Fix coming soon.

**Comment by alan-knight** [comment]
https://codereview.chromium.org/1410853005/

**Comment by terrylucas** [comment]
Alan made a 1.13 merge request for this fix yesterday

Wrap DomStringList, but don't wrap DomString[] as return types -
https://codereview.chromium.org/1410853005

This was merged in 1.13.0-dev.7.8
